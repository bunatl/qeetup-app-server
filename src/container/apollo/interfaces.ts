import { Context as DefaultContext } from 'apollo-server-core';
import { DataSources } from '../data-sources';

export interface Context extends DefaultContext {
    dataSources: DataSources;
}
