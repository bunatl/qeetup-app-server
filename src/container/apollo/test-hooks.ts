import { createTestClient } from 'apollo-server-testing';
import { createApolloServer } from '.';

/** Use this function for creating test client instances for Apollo server with default configuration. */
export const createApolloTestClient = () => {
    const server = createApolloServer();
    return createTestClient(server);
};
