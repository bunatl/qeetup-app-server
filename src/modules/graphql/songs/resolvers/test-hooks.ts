import { gql } from 'apollo-server';

export const SONG_FRAGMENT = gql`
    fragment SongFragment on Song {
        name
        artist
        cover
        description
        listens
        tags {
            value
            isImportant
        }
        audio
        isLiked
        comments {
            user {
                name
                avatar
                isArtist
            }
            text
        }
    }
`;
