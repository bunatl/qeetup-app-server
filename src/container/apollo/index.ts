import { ApolloServer } from 'apollo-server';
import { ConnectionContext } from 'subscriptions-transport-ws';
import { createDataSources } from '../data-sources';
import { schemaDirectives } from '../directives';
import { resolvers } from '../resolvers';
import { typeDefs } from '../schema';

export { Context } from './interfaces';

export const createApolloServer = () => {
    const dataSources = createDataSources();
    return new ApolloServer({
        typeDefs,
        resolvers,
        schemaDirectives,
        dataSources: () => dataSources,
        context: ({ connection }: { connection: ConnectionContext | null }) => {
            // Necessary to duplicate data sources for subscriptions, because of separate web-socket connection.
            if (connection) return { dataSources, webSockets: true };
            return { webSockets: false };
        },
    });
};
