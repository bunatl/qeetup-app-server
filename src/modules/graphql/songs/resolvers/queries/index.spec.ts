import { createApolloTestClient } from '@container/apollo/test-hooks';
import { gql } from 'apollo-server';

const ALL_SONGS = gql`
    query AllSongs {
        songs {
            name
            artist
            cover
            description
            listens
            tags {
                value
                isImportant
            }
            audio
            isLiked
            comments {
                user {
                    name
                    avatar
                    isArtist
                }
                text
            }
        }
    }
`;

const SEARCH_SONGS = gql`
    query SearchSongs($name: String!) {
        search(name: $name) {
            name
            artist
            cover
            description
            listens
            tags {
                value
                isImportant
            }
            audio
            isLiked
            comments {
                user {
                    name
                    avatar
                    isArtist
                }
                text
            }
        }
    }
`;

describe('Songs Queries', () => {
    it('should return all songs', async () => {
        expect.assertions(1);

        const { query } = createApolloTestClient();
        const { data: songs } = await query({ query: ALL_SONGS });

        expect(songs).toMatchSnapshot();
    });

    it('should find all songs with given name', async () => {
        expect.assertions(1);

        const { query } = createApolloTestClient();
        const { data: songs } = await query({ query: SEARCH_SONGS, variables: { name: 'Bad' } });

        expect(songs).toMatchSnapshot();
    });
});
