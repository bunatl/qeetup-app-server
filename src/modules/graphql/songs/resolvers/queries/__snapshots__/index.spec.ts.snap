// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Songs Queries should find all songs with given name 1`] = `
Object {
  "search": Array [
    Object {
      "artist": "September 87",
      "audio": "BadDreamBaby",
      "comments": null,
      "cover": "https://i.vimeocdn.com/video/526078435_1280x720.jpg",
      "description": "This song is by September 87 and features Dream Fiend.",
      "isLiked": true,
      "listens": 145202,
      "name": "Bad Dream Baby",
      "tags": Array [
        Object {
          "isImportant": true,
          "value": "New Synth Wave",
        },
        Object {
          "isImportant": true,
          "value": "ðŸ”¥ Popular",
        },
        Object {
          "isImportant": null,
          "value": "Electronic",
        },
        Object {
          "isImportant": null,
          "value": "Retro",
        },
        Object {
          "isImportant": null,
          "value": "Outrun",
        },
      ],
    },
  ],
}
`;

exports[`Songs Queries should return all songs 1`] = `
Object {
  "songs": Array [
    Object {
      "artist": "September 87",
      "audio": "BadDreamBaby",
      "comments": null,
      "cover": "https://i.vimeocdn.com/video/526078435_1280x720.jpg",
      "description": "This song is by September 87 and features Dream Fiend.",
      "isLiked": true,
      "listens": 145202,
      "name": "Bad Dream Baby",
      "tags": Array [
        Object {
          "isImportant": true,
          "value": "New Synth Wave",
        },
        Object {
          "isImportant": true,
          "value": "ðŸ”¥ Popular",
        },
        Object {
          "isImportant": null,
          "value": "Electronic",
        },
        Object {
          "isImportant": null,
          "value": "Retro",
        },
        Object {
          "isImportant": null,
          "value": "Outrun",
        },
      ],
    },
    Object {
      "artist": "The Midnight",
      "audio": "LosAngeles",
      "comments": null,
      "cover": "https://scontent-prg1-1.xx.fbcdn.net/v/t1.0-9/44883276_2160020650889409_8949487783479607296_n.jpg?_nc_cat=107&_nc_oc=AQnVcG20lk3-m112Suw1y55PovPGTdDmScc_ENTqVQOBrZTl-giFBV71MEzLVvjl8rbazDD0qK55rK6859UkL4AI&_nc_ht=scontent-prg1-1.xx&oh=0e884e2a8643364267b2d42968551401&oe=5E255286",
      "description": "Placeholder",
      "isLiked": true,
      "listens": 34000,
      "name": "Los Angeles",
      "tags": Array [
        Object {
          "isImportant": true,
          "value": "New Synth Wave",
        },
        Object {
          "isImportant": true,
          "value": "ðŸ”¥ Popular",
        },
        Object {
          "isImportant": null,
          "value": "Electronic",
        },
        Object {
          "isImportant": null,
          "value": "Retro",
        },
        Object {
          "isImportant": null,
          "value": "Outrun",
        },
      ],
    },
    Object {
      "artist": "FM-84",
      "audio": null,
      "comments": null,
      "cover": "https://i.redd.it/xvyfcvt7fvwz.jpg",
      "description": "This song is by September 87 and features Dream Fiend.",
      "isLiked": false,
      "listens": 32000,
      "name": "Running in the night",
      "tags": Array [
        Object {
          "isImportant": true,
          "value": "New Synth Wave",
        },
        Object {
          "isImportant": true,
          "value": "ðŸ”¥ Popular",
        },
        Object {
          "isImportant": null,
          "value": "Electronic",
        },
        Object {
          "isImportant": null,
          "value": "Retro",
        },
        Object {
          "isImportant": null,
          "value": "Outrun",
        },
      ],
    },
    Object {
      "artist": "MÃ¶tley CrÃ¼e",
      "audio": null,
      "comments": null,
      "cover": "https://townsquare.media/site/295/files/2019/02/MCscr.jpg?w=980&q=75",
      "description": "This song is by September 87 and features Dream Fiend.",
      "isLiked": true,
      "listens": 1030002,
      "name": "Kickstart my heart",
      "tags": Array [
        Object {
          "isImportant": null,
          "value": "Glam metal",
        },
        Object {
          "isImportant": true,
          "value": "ðŸ”¥ Popular",
        },
        Object {
          "isImportant": null,
          "value": "Heavy metal",
        },
        Object {
          "isImportant": null,
          "value": "Hard Rock",
        },
      ],
    },
    Object {
      "artist": "Pink Floyd",
      "audio": null,
      "comments": null,
      "cover": "https://images-na.ssl-images-amazon.com/images/I/81aTawcGdmL._SL1500_.jpg",
      "description": "Placeholder",
      "isLiked": false,
      "listens": 8398893,
      "name": "Speak To Me",
      "tags": Array [
        Object {
          "isImportant": true,
          "value": "Progressive rock",
        },
        Object {
          "isImportant": null,
          "value": "Rock",
        },
      ],
    },
    Object {
      "artist": "The Glitch Mob",
      "audio": null,
      "comments": null,
      "cover": "https://consequenceofsound.net/wp-content/uploads/2014/01/91vaxjhtjjl-_sl1500_.jpg?quality=80&w=807",
      "description": "Placeholder",
      "isLiked": true,
      "listens": 490000,
      "name": "Mind of a Beast",
      "tags": Array [
        Object {
          "isImportant": true,
          "value": "Glitch",
        },
        Object {
          "isImportant": null,
          "value": "Electronic rock",
        },
        Object {
          "isImportant": null,
          "value": "Dubstep",
        },
        Object {
          "isImportant": null,
          "value": "EDM",
        },
      ],
    },
  ],
}
`;
